### ISSUE TITLE
OS X build errors on master [phab:T102]

### ISSUE THREAD

**Issue by kythe** [google/kythe#982]
_Created by **benjyw** at 2016-02-10 16:33:45_: 

Following these instructions: https://www.kythe.io/2015/06/05/osx-support/ and building at a recent master HEAD.

Got this error during `bazel build //...`. 

```
INFO: Found 749 targets...
INFO: From Linking kythe/cxx/indexer/cxx/test:
ld: warning: option -noall_load is obsolete and being ignored
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: bazel-out/local_darwin-fastbuild/bin/kythe/cxx/indexer/cxx/test (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
ERROR: /Users/benjyw/src/kythe/kythe/cxx/indexer/cxx/BUILD:95:1: Linking of rule '//kythe/cxx/indexer/cxx:test' failed: clang failed: error executing command /Users/benjyw/src/kythe/tools/cpp/clang -o bazel-out/local_darwin-fastbuild/bin/kythe/cxx/indexer/cxx/test '-Wl,-rpath,$ORIGIN/../../../../_solib_darwin/' ... (remaining 86 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
```

Note that building at the v0.0.17 tag works fine, so I'm not stuck, but thought you might find this information useful.

-----------------------------

My versions:

Kernel:

```
[~/src/kythe][master]$ uname -a
Darwin xxxx.local 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64 
```

OS X:

```
[~/src/kythe][master]$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.3
BuildVersion:	15#212 
```

gcc/clang:

```
[~/src/kythe][master]$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin15.3.0
Thread model: posix
 ```

bazel:

```
[~/src/kythe][master]$ bazel version
Build label: 0.1.4-homebrew
Build target: bazel-out/local_darwin-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Sat Jan 23 12:15:20 2016 (1453551320)
Build timestamp: 1453551320
Build timestamp as int: 1453551320
```

kythe:

```
[~/src/kythe][master]$ git rev-parse HEAD
a10b8b927bceef37204a6fb5dac34d2f45c5950e
```

tool paths:

```
[~/src/kythe][master]$ ./setup_bazel.sh
Using GOROOT found at /usr/local/Cellar/go/1.5.3/libexec
Using node.js found at /usr/local/Cellar/node/0.10.33_1/bin/node
Using clang found at /usr/bin/clang
```

**Comment by kythe** [comment]
_**zarko** wrote at 2016-02-10 19:13:38_:

This is working for me.
```
bazel version
Build label: 0.1.5
Build target: bazel-out/local_darwin-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Feb 9 19:11:16 2016 (1455045076)
Build timestamp: 1455045076
Build timestamp as int: 1455045076
```
```
uname -a
Darwin XXX 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
```
```
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.3
BuildVersion:	15#212
```
```
gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1
Apple LLVM version 7.0.0 (clang-700.0.72)
Target: x86_64-apple-darwin15.3.0
Thread model: posix
```
```
git rev-parse HEAD
187331c92a4764b736657163428195730993ff71
```
```
./setup_bazel.sh
Using GOROOT found at /usr/local/go
No node.js installation found.
Using clang found at /usr/bin/clang
```

**Comment by kythe** [comment]
_**zarko** wrote at 2016-02-10 19:14:36_:

Note that both `build //...` and `test //...` are working at that commit.

**Comment by kythe** [comment]
_**benjyw** wrote at 2016-02-17 01:46:35_:

This still happens at current master ():



> $:[~/src/kythe][master]$ bazel build //...
> INFO: Waiting for response from Bazel server (pid 37121)...
> INFO: Found 756 targets...
> INFO: From Linking kythe/cxx/indexer/cxx/test:
> ld: warning: option -noall_load is obsolete and being ignored
> error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: bazel-out/local_darwin-fastbuild/bin/kythe/cxx/indexer/cxx/test (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
> ERROR: /Users/benjyw/src/kythe/kythe/cxx/indexer/cxx/BUILD:95:1: Linking of rule '//kythe/cxx/indexer/cxx:test' failed: clang failed: error executing command /Users/benjyw/src/kythe/tools/cpp/clang -o bazel-out/local_darwin-fastbuild/bin/kythe/cxx/indexer/cxx/test '-Wl,-rpath,$ORIGIN/../../../../_solib_darwin/' ... (remaining 86 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
> INFO: Elapsed time: 9.189s, Critical Path: 4.75s




Will try with bazel 0.1.5 and see if that helps any.

**Comment by kythe** [comment]
_**benjyw** wrote at 2016-02-17 01:59:42_:

That didn't help. Same error with bazel 0.1.5.

**Comment by kythe** [comment]
_**benjyw** wrote at 2016-02-18 00:09:18_:

I've submitted a code review that fixes this for me:

#1007

**Comment by kythe** [comment]
_**zarko** wrote at 2016-03-21 23:55:52_:

Fixed in #1008
